spec/policies/planned_disbursement_policy_spec.rb

```diff
@@ -1,103 +1,186 @@
+    end
+  end
 
-      context "when the parent is a third_party_project" do
-        let(:activity) { create(:third_party_project_activity, :with_report, organisation: user.organisation) }
-        it { is_expected.to permit_action(:update) }
-      end
+  context "when signed in as a Delivery partner user" do
+    let(:user) { create(:delivery_partner_user) }
 
-      context "when the user belongs to the authoring organisation" do
-        let(:planned_disbursement) { create(:planned_disbursement, parent_activity: activity) }
-        it { is_expected.to permit_action(:update) }
-      end
+    context "when the activity has no level" do
+      let(:activity) { CreateActivity.new(organisation_id: user.organisation.id).call }
 
-      context "when the user does NOT belong to the authoring organisation" do
-        let(:another_organisation) { create(:organisation) }
-        let(:activity) { create(:activity, organisation: another_organisation) }
-        let(:planned_disbursement) { create(:planned_disbursement, parent_activity: activity) }
-        it { is_expected.to forbid_action(:update) }
-      end
+      it { is_expected.to permit_action(:show) }
+
+      it { is_expected.to forbid_action(:create) }
+      it { is_expected.to forbid_action(:edit) }
+      it { is_expected.to forbid_action(:update) }
+      it { is_expected.to forbid_action(:destroy) }
     end
 
-    context "when the planned disbursement is associated to an approved report" do
-      let(:activity) { create(:project_activity, organisation: user.organisation) }
-      let(:report) { create(:report, :approved, organisation: activity.organisation, fund: activity.associated_fund) }
-      let(:planned_disbursement) { create(:planned_disbursement, parent_activity: activity, report: report) }
+    context "when the activity is a fund" do
+      let(:activity) { create(:fund_activity, organisation: user.organisation) }
+
+      it { is_expected.to permit_action(:show) }
+
+      it { is_expected.to forbid_action(:create) }
+      it { is_expected.to forbid_action(:edit) }
       it { is_expected.to forbid_action(:update) }
+      it { is_expected.to forbid_action(:destroy) }
     end
-  end
 
-  describe "#destroy?" do
-    let(:planned_disbursement) { create(:planned_disbursement, parent_activity: activity) }
-    it { is_expected.to forbid_action(:destroy) }
+    context "when the activity is a programme" do
+      let(:activity) { create(:programme_activity, organisation: user.organisation) }
+
+      it { is_expected.to permit_action(:show) }
+
+      it { is_expected.to forbid_action(:create) }
+      it { is_expected.to forbid_action(:edit) }
+      it { is_expected.to forbid_action(:update) }
+      it { is_expected.to forbid_action(:destroy) }
+    end
+
+    context "when the activity is a project" do
+      let(:activity) { create(:project_activity) }
+
+      context "and the activity does not belong to the users organiastion" do
```

Comment:
(Sorry for the picky comment!) There is a misspelling of `organisation` in the context description